ui/desktop/src/components/BaseChat.tsx

```diff
@@ -200,16 +201,25 @@ function BaseChatContent({
   // Reset recipe usage tracking when recipe changes
   useEffect(() => {
     if (recipeConfig?.title !== currentRecipeTitle) {
+      const previousTitle = currentRecipeTitle;
       setCurrentRecipeTitle(recipeConfig?.title || null);
-      setHasStartedUsingRecipe(false);
 
-      // Clear existing messages when a new recipe is loaded
-      if (recipeConfig?.title && recipeConfig.title !== currentRecipeTitle) {
+      // Only reset usage if we're switching between different recipes
+      // Don't reset if we're going from no recipe to a recipe (initial load)
+      // or if we already have messages (ongoing conversation)
+      if (previousTitle && recipeConfig?.title && previousTitle !== recipeConfig.title) {
```

Comment:
here and below, we already checked previousTitle != recipeConfig.title so that bit is always going to be true. So for this to not be true, either previousTitle or recipeConfig?title needs to be falsey. if that's the case we should check on that